CONTRIBUTING.md

```diff
@@ -1,19 +1,184 @@
 # Contributing
 
-## Installations
+## Local development
 
-If you don't already have some sort of local server, you should install the `docsify-cli` package. You can do so by running the following command:
+Clone the repo into your desired folder and `cd` into it.
 
+```bash
+git clone https://github.com/discordjs/guide.git && cd guide
 ```
-npm i -g docsify-cli
+
+If you don't already have some sort of local server, you should install the [docsify-cli package](https://github.com/QingWei-Li/docsify-cli). You can do so by running:
+
+```bash
+npm install -g docsify-cli
+```
+
+After that, simply run `docsify serve ./guide --open` inside the project's root directory to run a server at http://localhost:3000. The `--open` flag will make it automatically open it in your browser. If you need to use a different port, use the `--port` flag.
+
+### Linting
+
+Remember to always lint your edits/additions before making a commit, to ensure everything's lined up and consistent with the rest of the guide. We use ESLint and have a package.json script for linting both JS files and JS codeblocks inside Markdown files. Just install the dependencies and run the lint script.
+
+```bash
+npm install
+npm run lint
+```
+
+#### Caveats
+
+There might come a time where a snippet will contain a parse error, and ESLint won't be able to lint it properly. For example:
+
+<!-- eslint-skip -->
+
+```js
+const sent = await message.channel.send('Hi!');
+console.log(sent.content)
 ```
 
-After that, you can simply run `docsify serve guide` to run a server at `localhost:3000`.
+ESLint would error with `Parsing error: Unexpected token message` instead of letting you know that you're missing a semicolon. In this case, it's because of the use of `await` outside of an async function. In situations like this, after you've fixed any obvious errors, you can add an `<!-- eslint-skip -->` comment above the codeblock to have it ignored entirely by ESLint when running the lint script. 
 
-## Adding Pages
+## Adding pages
 
-To add a new page to the guide, simply create a `file-name.md` file inside the folder of your choice. If you want to link to `/dir/some-tutorial`, you would create a `some-tutorial.md` file inside a `dir` folder. Docsify will pick up on it and set up the routing appropriately.
+To add a new page to the guide, create a `file-name.md` file inside the folder of your choice. If you want to link to `/dir/some-tutorial`, you would create a `some-tutorial.md` file inside a `dir` folder. [Docsify](https://github.com/QingWei-Li/docsify/) will pick up on it and set up the routing appropriately.
```

Comment:
Trailing slash on the docsify link when there isn't one up there. This may possibly be the smallest nitpick :mmLol:.